src/components/appointments/ChooseProductStep.js

```diff
@@ -1,77 +1,134 @@
-                          description="Appointments: remove product/service button text"
-                          defaultMessage="Remove"
-                        />
-                      </Button>
-                    </ToolbarList>
-                  </Toolbar>
-                )}
-              </div>
-            ))}
-          </div>
+                  <Product namePrefix="products" index={index} />
+
+                  {numProducts > 1 && (
+                    <Toolbar modifiers={['reverse']}>
+                      <ToolbarList>
+                        <Button
+                          type="button"
+                          variant="danger"
+                          onClick={() => arrayHelpers.remove(index)}
+                        >
+                          <FormattedMessage
+                            description="Appointments: remove product/service button text"
+                            defaultMessage="Remove"
+                          />
+                        </Button>
+                      </ToolbarList>
+                    </Toolbar>
+                  )}
+                </div>
+              ))}
+            </div>
 
-          <div className={getBEMClassName('editgrid__add-button')}>
-            <Button
-              type="button"
-              variant="primary"
-              onClick={() => arrayHelpers.push({product: '', amount: 1})}
-            >
-              <FAIcon icon="plus" />
-              <FormattedMessage
-                description="Appointments: add additional product/service button text"
-                defaultMessage="Add another product"
-              />
-            </Button>
+            <div className={getBEMClassName('editgrid__add-button')}>
+              <Button
+                type="button"
+                variant="primary"
+                onClick={() => arrayHelpers.push({product: '', amount: 1})}
+              >
+                <FAIcon icon="plus" />
+                <FormattedMessage
+                  description="Appointments: add additional product/service button text"
+                  defaultMessage="Add another product"
+                />
+              </Button>
+            </div>
           </div>
-        </div>
-      )}
-    </FieldArray>
+        )}
+      </FieldArray>
+
+      <ButtonsToolbar
+        literals={{
+          nextText: {
+            resolved: intl.formatMessage({
+              description: 'Appointments products step: next step text',
+              defaultMessage: 'Confirm products',
+            }),
+          },
+        }}
+        canSubmitStep={isStepValid(values)}
+        canSubmitForm={SUBMISSION_ALLOWED.yes}
+        canSuspendForm={false}
+        isAuthenticated={false} // TODO -> if authenticated (for prefill), logout must be shown
+        isLastStep={false}
+        isCheckingLogic={false}
+        loginRequired={false}
+        onFormSave={() => {}}
+        onLogout={() => {}}
+      />
```

Comment:
*probably* better to use the primitives here instead of passing in a lot of irrelevant props. All we need is simple back links and submit buttons.